packages/org.openapis.v3/OAuthFlow.pkl

```diff
@@ -0,0 +1,61 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+/// Flows contain configuration details for a supported OAuth Flow.
+module org.openapis.v3.OAuthFlow
+
+/// Common OAuth flow fields.
+abstract class Flow {
+
+  /// The URL to be used for obtaining refresh tokens. This MUST be in the form of a URL.
+  refreshUrl: Uri?
+
+  /// The available scopes for the OAuth2 security scheme.
+  ///
+  /// A map between the scope name and a short description for it. The map MAY be empty.
+  scopes: Mapping<String, String>
+}
+
+/// Flow with authorization URL.
+class WithAuthorizationUrl extends Flow {
+
+  /// The authorization URL to be used for this flow.
+  ///
+  /// This MUST be in the form of a URL.
+  authorizationUrl: Uri
+}
+
+/// Flow with token URL.
+class WithTokenUrl extends Flow {
+
+  /// The token URL to be used for this flow.
+  ///
+  /// This MUST be in the form of a URL.
+  tokenUrl: Uri
+}
+
+/// Flow with both token and authorization URLs.
+class WithTokenAndAuthorizationUri extends Flow {
+
+  /// The authorization URL to be used for this flow.
+  ///
+  /// This MUST be in the form of a URL.
+  authorizationUrl: Uri
+
+  /// The token URL to be used for this flow.
+  ///
+  /// This MUST be in the form of a URL.
+  tokenUrl: Uri
+}
```

Comment:
Let's consolidate this into module `OAuthFlows` and delete this module (see suggestion there)